webapp/src/scenes/app/components/TermsOfServiceDialog.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import MarkdownIt from 'markdown-it'
+import md5 from 'crypto-js/md5';
+
+import CustomDialog from '../../../components/CustomDialog';
+import { administrationApi, getDefaultApiCallback } from '../../../services/client/ml-lab-api';
+
+const md = new MarkdownIt();
+
+class TermsOfServiceDialog extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      open: false,
+      tosText: "Initial TOS"
+    };
+    this.requestTOS();
+    this.acceptedTosList = JSON.parse(localStorage.getItem("acceptedTOS") || '[]');
+  }
+
+  requestTOS() {
+    administrationApi.getLabInfo(null,
+      getDefaultApiCallback(
+        ({ result }) => {
+          let tosText = result.data.termsOfService;
+          let tosDisabled = tosText == "ToS disabled\n"
```

Comment:
I think it would be better to compare to an empty string to not make any assumptions on the content of the ToS.